pycheribuild/projects/disk_image.py

```diff
@@ -975,7 +975,7 @@ def add_unlisted_files_to_metalog(self):
         self.mtree.add_dir("var/db", print_status=self.config.verbose)
         self.mtree.add_dir("var/empty", print_status=self.config.verbose)
 
-        if self.is_x86 or self.compiling_for_aarch64(include_purecap=True):
+        if self.include_boot and (self.is_x86 or self.compiling_for_aarch64(include_purecap=True)):
```

Comment:
This does mean that kgdb and dtrace will be sadder (though without the .full kernel maybe one or both already are), but that's not why this code was here, it's here based on whether you're direct-booting kernels or not, so that's a separate issue.

Reply:
At a glance, `/boot/kernel/kernel` does contain a `.SUNW_ctf` and `.symtab` section, so I could see a reason to want those even on some `self.include_boot = False` builds.  Sounds like the MFS disk image classes might want a build flag.